core/spec/models/spree/order/state_machine_spec.rb

```diff
@@ -113,9 +113,18 @@
         end
       end
 
-      it "should automatically refund all payments" do
+      context "with a pending payment" do
```

Comment:
If I understand correctly, the tested behavior depends on the underlying payment method. Do these tests make sense here?

Reply:
You are right, this should probably go in the specific payment method specs. And we should just test that the outer interface method is called here. Let me see if I can do the change.

Reply:
Again, you were right. This behavior is already tested in [order specs](https://github.com/solidusio/solidus/blob/36340a8785294b7287b03599232be8bf44de548e/core/spec/models/spree/order_spec.rb#L43-L125) and [payment cancellation specs](https://github.com/solidusio/solidus/blob/36340a8785294b7287b03599232be8bf44de548e/core/spec/models/spree/payment/cancellation_spec.rb). This change gives us the opportunity to remove the duplication.

In [payment method specs](https://github.com/solidusio/solidus/blob/36340a8785294b7287b03599232be8bf44de548e/core/spec/models/spree/payment_method_spec.rb#L223-L329) instead, we now have the opportunity to just use the bogus credit card instead of the anonymous classes. The downside is that we'll connect these bogus implementations to the tests, but I think it also has an advantage: if we make some changes that make these specs fail, it most likely will be the same for other payment gateways with the same interface, and at least we will notice immediately.

Please re-review now, commit-by-commit should be easier.